static/src/javascripts/projects/common/modules/commercial/sticky-top-banner.js

```diff
@@ -46,26 +48,17 @@ define([
     // Rubicon ads are loaded via DFP like all other ads, but they can
     // render themselves again at any time
     var newRubiconAdHeightPromise = new Promise(function (resolve) {
-        window.addEventListener('message', function (event) {
-            var data;
-            // other DFP events get caught by this listener, but if they're not json we don't want to parse them or use them
-            try {
-                data = JSON.parse(event.data);
-            } catch (e) {/**/}
-
-            if (data) {
-                var $iframe = getAdIframe();
-                var isRubiconAdEvent = data.type === 'set-ad-height';
-                var isEventForTopAdBanner = isRubiconAdEvent && data.value.id === $iframe[0].id;
-
-                if (isRubiconAdEvent && isEventForTopAdBanner) {
-                    fastdom.read(function () {
-                        var padding = parseInt($adBannerInner.css('padding-top'))
-                            + parseInt($adBannerInner.css('padding-bottom'));
-                        var clientHeight = parseInt(data.value.height) + padding;
-                        resolve(clientHeight);
-                    });
-                }
+        messenger.register('set-ad-height', function onSetAdHeight(data) {
+            var $iframe = getAdIframe();
```

Comment:
addendum: so indeed, the `name` attribute is accessible as `window.name` and the good news is that DFP uses it. yiihaaa!